### PR TITLE
Book and label features

### PR DESCRIPTION
**Pull request by kamzzy**
###  Commit details

- Created a Book class in a separate .rb file.
- Create a Label class with an association to the Item class (in a separate .rb file).
- can_be_archived method implemented
- add_item method implemented
- data is saved in JSON files
- added tests for book methods
- added tests for label method

### CODE REVIEW

File: classes/book.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require '../item'
+class Book < Item
+  def initialize(publisher, cover_state, publish_date = Time.year)
```
Comment (RWUBAKWANAYO) [inline classes/book.rb:3]: - [ ] Replace Time method
```suggestion
  def initialize(publisher, cover_state, publish_date = Time.now)
```

File: classes/book.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    super(publish_date)
```
Comment (RWUBAKWANAYO) [inline classes/book.rb:4]: - [ ] Extend `archived` from super method
 ```suggestion
   super(publish_date, archived: archived, id: id)
```

File: classes/book.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    @publisher = publisher
+    @cover_state = cover_state
+  end
+
+  def can_be_archived?
+    true if Time.now.year - @publish_date.year > 10 || @cover_state == 'bad'
```
Comment (RWUBAKWANAYO) [inline classes/book.rb:10]: - [ ] `archived`canbe modified depending on property from a parent class or cover state.
```suggestion
super || cover_state == 'bad'
```

**Review by RWUBAKWANAYO — CHANGES_REQUESTED** [review]
Hi, @kamzzy  👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following: 
- Nicely structured documentation.✔️
- Clean Ruby code and style guide.✔️
- No linter errors.✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏
***
- [ ] Remember to check if your classes and methods work as expected.

File: classes/book.rb (hunk 1 above)
Comment (kamzzy) [reply classes/book.rb:3]: publish_date is inherited from the item class so the change of publish_date has to come from there

File: classes/book.rb (hunk 2 above)
Comment (kamzzy) [reply classes/book.rb:4]: can you explain why archived and id should be  extended its not part of the uml diagram

File: classes/book.rb (hunk 3 above)
Comment (kamzzy) [reply classes/book.rb:10]: I don't think archived is needed, all that is needed for can_be_archived is publish_date, not archived

Comment (kamzzy) [reply classes/book.rb:10]: can you explain better what you mean

**Comment by kamzzy** [comment]
@RWUBAKWANAYO  I have added tests and made  some modifications

**Review by RWUBAKWANAYO — COMMENTED** [review]
Hi @kamzzy, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯

- No linter errors
- Good code design and implementation. ✔️
- Add all the requirements. ✔️



- [ ] Remember to check if your classes and methods work as expected.
Cheers and Happy coding!👏👏👏

**Pull request merged**